src/main/java/com/statful/client/framework/springboot/registry/StatfulMetricRegistry.java

```diff
@@ -0,0 +1,190 @@
+package com.statful.client.framework.springboot.registry;
+
+import com.statful.client.domain.api.StatfulClient;
+import com.statful.client.domain.api.Tags;
+import io.micrometer.core.instrument.*;
+import io.micrometer.core.instrument.config.MeterFilter;
+import io.micrometer.core.instrument.distribution.HistogramSnapshot;
+import io.micrometer.core.instrument.step.StepMeterRegistry;
+import io.micrometer.core.instrument.step.StepRegistryConfig;
+import org.apache.logging.log4j.util.Strings;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static io.micrometer.core.instrument.util.StringUtils.isNotEmpty;
+
+public class StatfulMetricRegistry extends StepMeterRegistry {
+
+    private StatfulClient statfulClient;
+
+    private StatfulMetricsProperties properties;
+
+    public StatfulMetricRegistry(StepRegistryConfig config, StatfulClient statfulClient, Clock clock, StatfulMetricsProperties properties) {
+        super(config, clock);
+
+        this.statfulClient = statfulClient;
+        this.properties = properties;
+
+        applyCustomFilter();
+
+        start();
```

Comment:
This is deprecated, should be receiving a ThreadFactory. Maybe do `start(Executors.defaultThreadFactory());`for this constructor, and add a new constructor that receives the needed ThreadFactory.